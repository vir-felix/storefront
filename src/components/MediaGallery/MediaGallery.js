import React, { Component } from "react";
import PropTypes from "prop-types";
import Grid from "material-ui/Grid";
import withStyles from "material-ui/styles/withStyles";
import { inject, observer } from "mobx-react";
import MediaGalleryItem from "components/MediaGalleryItem";

const styles = (theme) => ({
  root: {
    width: "100%"
  },
  featured: {
    display: "flex",
    justifyContent: "center",
    height: "500px",
    maxHeight: "500px",
    overflow: "hidden",
    backgroundColor: theme.palette.common.white,
    marginBottom: theme.spacing.unit,
    [theme.breakpoints.up("xs")]: {
      maxHeight: "300px"
    },
    [theme.breakpoints.up("sm")]: {
      maxHeight: "300px"
    },
    [theme.breakpoints.up("md")]: {
      maxHeight: "500px"
    },
    [theme.breakpoints.up("lg")]: {
      maxHeight: "500px"
    }
  },
  featuredImage: {
    flex: 0,
    height: "100%"
  }
});

/**
 * Product detail media gallery
 * @class ProductDetailMediaGallery
 */
@withStyles(styles, { withTheme: true })
@inject("uiStore")
@observer
class MediaGallery extends Component {
  static propTypes = {
    /**
     * CSS Class names
     */
    classes: PropTypes.object,

    /**
     * Media items
     */
    mediaItems: PropTypes.arrayOf(PropTypes.object),

    /**
     * MUI Theme
     */
    theme: PropTypes.object,

    /**
     * UIStore
     */
    uiStore: PropTypes.object
  }

  static defaultProps = {
    mediaItems: []
  };

  state = { featuredMediaIndex: 0 };

  /**
   * @name handleMediaItemClick
   * @param {SyntheticEvent} event Event
   * @param {Object} media The `media` prop of the MediaGalleryItem that was clicked
   * @param {Number} index The `index` prop of the MediaGalleryItem that was clicked
   * @returns {undefined} Nothing
   */
  handleMediaItemClick = (event, media, index) => {
    this.setState({ featuredMediaIndex: index });
  };

  renderFeaturedImage() {
    const { classes, mediaItems, uiStore } = this.props;
    const { publicRuntimeConfig } = uiStore.appConfig;
    const featuredMedia = mediaItems[this.state.featuredMediaIndex];
    const mediaUrl = featuredMedia && featuredMedia.URLs && featuredMedia.URLs.large;

    return (
      <img
        className={classes.featuredImage}
        src={`${publicRuntimeConfig.externalAssetsUrl}${mediaUrl || publicRuntimeConfig.placeholderImageUrls.galleryFeatured}`}
        alt=""
      />
    );
  }

  render() {
    const { classes, mediaItems, theme } = this.props;

    return (
      <Grid container className={classes.root}>
        <Grid item xs={12} sm={12}>
          <div className={classes.featured}>
            {this.renderFeaturedImage()}
          </div>

          <Grid container spacing={theme.spacing.unit}>
            {mediaItems.map((media, index) => (
              <Grid
                item
                key={index}
                xs={3}
                sm={2}
              >
                <MediaGalleryItem
                  index={index}
                  media={media}
                  onClick={this.handleMediaItemClick}
                />
              </Grid>
            ))}
          </Grid>
        </Grid>
      </Grid>
    );
  }
}

export default MediaGallery;
