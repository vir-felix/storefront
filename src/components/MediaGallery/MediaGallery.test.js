import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { Provider } from "mobx-react";
import MediaGallery from "./MediaGallery";

const uiStore = {
  appConfig: {
    publicRuntimeConfig: {
      externalAssetsUrl: ""
    }
  }
};

const mediaItems = [
  {
    toGrid: 1,
    priority: 0,
    productId: "cmVhY3Rpb24vcHJvZHVjdDpCQ1RNWjZIVHhGU3BwSkVTaw==",
    variantId: "cmVhY3Rpb24vcHJvZHVjdDo2cWlxUHdCa2VKZHRkUWM0Rw==",
    URLs: {
      large: "/assets/files/Media/P3MkhDWadHoDNsFPE/large/I0DhYYC.jpg",
      medium: "/assets/files/Media/P3MkhDWadHoDNsFPE/medium/I0DhYYC.jpg",
      original: "/assets/files/Media/P3MkhDWadHoDNsFPE/image/I0DhYYC.jpg",
      small: "/assets/files/Media/P3MkhDWadHoDNsFPE/small/I0DhYYC.png",
      thumbnail: "/assets/files/Media/P3MkhDWadHoDNsFPE/thumbnail/I0DhYYC.png"
    }
  }, {
    toGrid: 1,
    priority: 0,
    productId: "cmVhY3Rpb24vcHJvZHVjdDpCQ1RNWjZIVHhGU3BwSkVTaw==",
    variantId: "cmVhY3Rpb24vcHJvZHVjdDo2cWlxUHdCa2VKZHRkUWM0Rw==",
    URLs: {
      large: "/assets/files/Media/P3MkhDWadHoDNsFPE/large/I0DhYYC.jpg",
      medium: "/assets/files/Media/P3MkhDWadHoDNsFPE/medium/I0DhYYC.jpg",
      original: "/assets/files/Media/P3MkhDWadHoDNsFPE/image/I0DhYYC.jpg",
      small: "/assets/files/Media/P3MkhDWadHoDNsFPE/small/I0DhYYC.png",
      thumbnail: "/assets/files/Media/P3MkhDWadHoDNsFPE/thumbnail/I0DhYYC.png"
    }
  }
];

test("basic snapshot", () => {
  const component = renderer.create((
    <Provider uiStore={uiStore}>
      <MediaGallery mediaItems={mediaItems} />
    </Provider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
