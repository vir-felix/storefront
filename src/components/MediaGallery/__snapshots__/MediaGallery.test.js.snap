// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<div
  className="MuiGrid-container-4 MediaGallery-root-1"
>
  <div
    className="MuiGrid-item-5 MuiGrid-grid-xs-12-42 MuiGrid-grid-sm-12-55"
  >
    <div
      className="MediaGallery-featured-2"
    >
      <div
        className="Img-imgWrapper-95 "
      >
        <img
          alt=""
          className="Img-img-97 Img-imgLoading-99 "
          src="/resources/placeholder.gif"
        />
      </div>
    </div>
    <div
      className="MuiGrid-container-4 MuiGrid-spacing-xs-8-25"
    >
      <div
        className="MuiGrid-item-5 MuiGrid-grid-xs-3-33 MuiGrid-grid-sm-2-45"
      >
        <button
          className="MuiButtonBase-root-102 MediaGalleryItem-root-101"
          disabled={undefined}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <div
            className="Img-imgWrapper-95 "
          >
            <img
              alt=""
              className="Img-img-97 Img-imgLoading-99 "
              src="/resources/placeholder.gif"
            />
          </div>
          <span
            className="MuiTouchRipple-root-105"
          />
        </button>
      </div>
      <div
        className="MuiGrid-item-5 MuiGrid-grid-xs-3-33 MuiGrid-grid-sm-2-45"
      >
        <button
          className="MuiButtonBase-root-102 MediaGalleryItem-root-101"
          disabled={undefined}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <div
            className="Img-imgWrapper-95 "
          >
            <img
              alt=""
              className="Img-img-97 Img-imgLoading-99 "
              src="/resources/placeholder.gif"
            />
          </div>
          <span
            className="MuiTouchRipple-root-105"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
